it/src/main/java/com/google/cloud/teleport/it/spanner/DefaultSpannerManager.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.spanner;
+
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.InstanceAdminClient;
+import com.google.cloud.spanner.InstanceConfigId;
+import com.google.cloud.spanner.InstanceId;
+import com.google.cloud.spanner.InstanceInfo;
+import com.google.cloud.spanner.KeySet;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.CharMatcher;
+import com.google.common.hash.HashFunction;
+import com.google.common.hash.Hashing;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+public class DefaultSpannerManager implements SpannerResourceManager {
+  private InstanceId instanceId;
+  private DatabaseId databaseId;
+
+  private final Spanner spanner;
+  private final InstanceAdminClient instanceAdminClient;
+  private final DatabaseAdminClient databaseAdminClient;
+  private DatabaseClient databaseClient;
+
+  private DefaultSpannerManager(Builder builder) {
+    this(SpannerOptions.newBuilder().setProjectId(builder.projectId).build().getService());
+    this.instanceId = builder.instanceId;
+    this.databaseId = builder.databaseId;
+    databaseClient = null;
+  }
+
+  @VisibleForTesting
+  DefaultSpannerManager(Spanner spanner) {
+    this.spanner = spanner;
+    this.instanceAdminClient = spanner.getInstanceAdminClient();
+    this.databaseAdminClient = spanner.getDatabaseAdminClient();
+  }
+
+  public static Builder builder(String testId, String projectId, String region) {
+    return new Builder(testId, projectId, region);
+  }
+
+  @Override
+  public void createTable(String statement) throws Exception {
+    if (spanner.isClosed()) {
+      throw new Exception("Manager has cleaned up all resources and cannot complete operation.");
```

Comment:
Should i make specific exception class in another file, maybe SpannerManagerExceptions, instead of throwing a general exception?